src/screens/wallet/screen/walletScreen.tsx

```diff
@@ -91,31 +77,52 @@ const WalletScreen = ({ navigation }) => {
       dispatch(resetWalletData());
       _refetchData();
     }
+    _updateSelectedCoinsDataFromProfileJsonMeta();
   }, [currency, currentAccount]);
 
   useEffect(() => {
     _fetchPriceHistory();
-  }, [selectedCoins])
+  }, [selectedCoins]);
 
+  //actions
+  const createEngineTokensDataFromProfileJsonMeta = (engineTokensData) => {
+    return engineTokensData.map((item) => ({
+      id: item,
+      symbol: item,
+      isEngine: true,
+      notCrypto: false,
+    }));
+  };
 
+  const _updateSelectedCoinsDataFromProfileJsonMeta = () => {
+    const currSelectedEngineTokens = selectedCoins.filter(
+      (item) => !DEFAULT_ASSETS.some((defaultCoin) => defaultCoin.id === item.id),
+    );
+    if (
+      currentAccount.about?.profile?.tokens &&
+      currentAccount.about?.profile?.tokens?.engine.length !== currSelectedEngineTokens.length // check if current selected engine tokens differ from profile json meta
```

Comment:
not a good comparison logic...see below example invalidating this logic... instead stringify might be a good option here.

![Screenshot 2023-01-24 at 5 28 43 PM](https://user-images.githubusercontent.com/6298342/214291905-b1116af7-9f1c-475a-a70c-c97779bb4131.png)